src/directives/NgGrid.ts

```diff
@@ -1245,4 +1264,76 @@ export class NgGrid implements OnInit, DoCheck, OnDestroy {
 	private _emitOnItemChange() {
 		this.onItemChange.emit(this._items.map((item: NgGridItem) => item.getEventOutput()));
 	}
+
+	private _defineListeners(): void {
+		const element = this._ngEl.nativeElement;
+		const eventOptions = { passive: true };
+
+		this._documentMousemove$ = Observable.fromEvent(document, 'mousemove', eventOptions);
+		this._documentMouseup$ = Observable.fromEvent(document, 'mouseup', eventOptions);
+		this._mousedown$ = Observable.fromEvent(element, 'mousedown', eventOptions);
+		this._mousemove$ = Observable.fromEvent(element, 'mousemove', eventOptions);
+		this._mouseup$ = Observable.fromEvent(element, 'mouseup', eventOptions);
+		this._touchstart$ = Observable.fromEvent(element, 'touchstart', eventOptions);
+		this._touchmove$ = Observable.fromEvent(element, 'touchmove', eventOptions);
+		this._touchend$ = Observable.fromEvent(element, 'touchend', eventOptions);
+		this._resize$ = Observable.fromEvent(window, 'resize');
+	}
+	
+	private _enableListeners(): void {
+		if (this._enabledListener) {
+			return;
+		}
+
+		this._enableMouseListeners();
+
+		if (this._isTouchDevice()) {
+			this._enableTouchListeners();
+		}
+ 
+		this._enabledListener = true;
+	}
+
+	private _disableListeners(): void {
+		this._subscriptions.forEach((subs: Subscription) => subs.unsubscribe());
+		this._enabledListener = false;
+	}
+
+	private _isTouchDevice(): boolean {
+		return 'ontouchstart' in window || navigator.maxTouchPoints > 0;
+	};
+	
+	private _enableTouchListeners(): void {
+		const touchstartSubs = this._touchstart$.subscribe((e: TouchEvent) => this.mouseDownEventHandler(e));
+		const touchmoveSubs = this._touchmove$.subscribe((e: TouchEvent) => this.mouseMoveEventHandler(e));
+		const touchendSubs = this._touchend$.subscribe((e: TouchEvent) => this.mouseUpEventHandler(e));
+
+		this._subscriptions.push(
+			touchstartSubs,
+			touchmoveSubs,
+			touchendSubs
+		);
+	}
+
+	private _enableMouseListeners(): void {
+		const documentMousemoveSubs = this._documentMousemove$.subscribe((e: MouseEvent) => this.mouseMoveEventHandler(e));
+		const documentMouseupSubs = this._documentMouseup$.subscribe((e: MouseEvent) => this.mouseUpEventHandler(e));
+		const mousedownSubs = this._mousedown$.subscribe((e: MouseEvent) => this.mouseDownEventHandler(e));
+		const mousemoveSubs = this._mousemove$.subscribe((e: MouseEvent) => this.mouseMoveEventHandler(e));
+		const mouseupSubs = this._mouseup$.subscribe((e: MouseEvent) => this.mouseUpEventHandler(e));
+
+		this._subscriptions.push(
+			documentMousemoveSubs,
+			documentMouseupSubs,
+			mousedownSubs,
+			mousemoveSubs,
+			mouseupSubs
+		);
+	}
+
+	private _enableResizeListeners(): void {
+		const resizeSubs = this._resize$.subscribe((e: MouseEvent) => this.resizeEventHandler(e));
+
+		this._subscriptions.push(resizeSubs);

```

Comment:
Should this be a standalone listener, rather than kept in the subscriptions method? Otherwise, it would be unsubscribed if the drag/resize config is not used

Reply:
Yes, you are right. I was trying to manage all listeners in the same way but since this listener must be always loaded I guess standalone listener is the best option. I will update it ASAP.